test/hotspot/jtreg/compiler/intrinsics/TestIntegerDivMod.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @summary Test x86_64 intrinsic for divideUnsigned() and remainderUnsigned() methods for Integer
+* @requires os.arch=="amd64" | os.arch=="x86_64"
+* @library /test/lib /
+* @run driver compiler.intrinsics.TestIntegerDivMod
+*/
+
+package compiler.intrinsics;
+import compiler.lib.ir_framework.*;
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+
+
+public class TestIntegerDivMod {
+    private final int BUFFER_SIZE = 1024;
+    private int[] dividends = new int[BUFFER_SIZE];
+    private int[] divisors = new int[BUFFER_SIZE];
+    private int[] quotients =  new int[BUFFER_SIZE];
+    private int[] remainders =  new int[BUFFER_SIZE];
+    private RandomGenerator rng;
+
+    public static void main(String args[]) {
+        TestFramework.run(TestIntegerDivMod.class);
+    }
+
+    public TestIntegerDivMod() {
+        rng = RandomGeneratorFactory.getDefault().create(0);
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            dividends[i] = rng.nextInt();
+            divisors[i] = rng.nextInt();
+        }
```

Comment:
I don't trust RND to generate corner cases.
Please, add cases here and in TestLongDivMod.java for MAX, MIN, 0.

Reply:
You are right. Using an updated corner cases test revealed divide by zero crash which was fixed. Please see the updated jtreg tests inspired by unsigned divide/remainder tests in test/jdk/java/lang/Long/Unsigned.java and test/jdk/java/lang/Integer/Unsigned.java.